app/src/org/odk/collect/android/tasks/FormLoaderTask.java

```diff
@@ -329,28 +343,34 @@ public FormDef deserializeFormDef(File formDef) {
 
 
     /**
-     * Write the FormDef to the file system as a binary blog.
+     * Write the FormDef to the file system as a binary blob.
      * 
      * @param filepath path to the form file
+     * @throws IOException 
      */
-    public void serializeFormDef(FormDef fd, String filepath) {
-        // calculate unique md5 identifier
+    public void serializeFormDef(FormDef fd, String filepath) throws IOException {
+        // calculate unique md5 identifier for this form
         String hash = FileUtils.getMd5Hash(new File(filepath));
         File formDef = new File(Collect.CACHE_PATH + "/" + hash + ".formdef");
 
-        // formdef does not exist, create one.
+        // create a serialized form file if there isn't already one at this hash
         if (!formDef.exists()) {
-            FileOutputStream fos;
+            FileOutputStream fos = null;
             try {
                 fos = new FileOutputStream(formDef);
                 DataOutputStream dos = new DataOutputStream(fos);
                 fd.writeExternal(dos);
                 dos.flush();
                 dos.close();
-            } catch (FileNotFoundException e) {
-                e.printStackTrace();
-            } catch (IOException e) {
-                e.printStackTrace();
+            } finally {
+                //make sure we clean up the stream
+                if(fos != null) {
+                    try {
+                        fos.close();
+                    } catch (IOException e) {
+                        // there was already an error, this one is just cruft
```

Comment:
Not sure I follow the logic of this try/catch
If we are already not catching IOException and letting it bubble up, why wrap fos.close() in a try? Wouldn't it suffice to just close the stream and let any exceptions propagate up the call stack?